tests/ui/linking/link-self-contained-unstable.rs

```diff
@@ -0,0 +1,17 @@
+// Checks that values for `-Clink-self-contained` other than the blanket enable/disable and
+// `+/-linker` require `-Zunstable-options`.
+
+//@ revisions: crto libc unwind sanitizers mingw
+//@ [crto] compile-flags: -Clink-self-contained=+crto
+//@ [libc] compile-flags: -Clink-self-contained=-libc
+//@ [unwind] compile-flags: -Clink-self-contained=+unwind
+//@ [sanitizers] compile-flags: -Clink-self-contained=-sanitizers
+//@ [mingw] compile-flags: -Clink-self-contained=+mingw
+
+fn main() {}
+
+//[crto]~? ERROR only `-C link-self-contained` values `y`/`yes`/`on`/`n`/`no`/`off`/`-linker`/`+linker` are stable, the `-Z unstable-options` flag must also be passed to use the unstable values
```

Comment:
```suggestion
//[crto]~? ERROR only `-C link-self-contained` values `y`/`yes`/`on`/`n`/`no`/`off`/`-linker`/`+linker` are stable
```
Nit: copypasting the full message is not necessary, this message is mostly for people (and for double checking).
I personally usually skip the "extra" parts of error message here - how to fix the error, the list of possible options, etc.
(Same for some other tests below.)

Reply:
Done as a separate commit to make review easier.